templates/rustdoc/navigation.html

```diff
@@ -0,0 +1,248 @@
+
+                                                        <li class="pure-menu-item">
+                                                            {# If the release has been yanked and failed to build, display a warning #}
+                                                            {%- if release.yanked and release.build_status -%}
+                                                                <a href="{{ release_url }}" class="pure-menu-link warn" title="{{ release_name }} is yanked">
+                                                                    <i class="fa fa-fw fa-warning"></i> {{ release.version }}
+                                                                </a>
+
+                                                            {# If the release is yanked but built, display an warning #}
+                                                            {%- elif release.yanked and not release.build_status -%}
+                                                                <a href="{{ release_url }}" class="pure-menu-link warn"
+                                                                    title="{{ release_name }} is yanked and docs.rs failed to build it">
+                                                                    <i class="fa fa-fw fa-warning"></i> {{ release.version }}
+                                                                </a>
+
+                                                            {# If the release failed to build, display a warning #}
+                                                            {%- elif release.build_status -%}
+                                                                <a href="{{ release_url }}" class="pure-menu-link warn" title="docs.rs failed to build {{ release_name }}">
+                                                                    <i class="fa fa-fw fa-warning"></i> {{ release.version }}
+                                                                </a>
+
+                                                            {# Otherwise just display the version #}
+                                                            {%- else -%}
+                                                                <a href="{{ release_url }}" class="pure-menu-link">
+                                                                    {{ release.version }}
+                                                                </a>
+                                                            {%- endif -%}
+                                                        </li>
+                                                    {%- endfor -%}
+                                                </ul>
+                                            </div>
+                                        </li>
+                                    </ul>
+                                </div>
+                            </div>
+                        </div>
+                    </li>
```

Comment:
> > Ditto on the repeated template, it would be great to factor it out.

> I think this is the same template as the last comment?

Right, it's still duplicated.